From 6d842b3dea0c4697a53d5de17b4eef0a367e8a9d Mon Sep 17 00:00:00 2001
From: Enric Balletbo i Serra <eballetbo@iseebcn.com>
Date: Fri, 2 Sep 2011 12:59:19 +0200
Subject: [PATCH 13/44] OMAP3: igep0020: Fix double registration of omapdss device

This patch fixes double registration of omapdss device

[    0.208343] ------------[ cut here ]------------
[    0.208404] WARNING: at fs/sysfs/dir.c:455 sysfs_add_one+0x90/0xb8()
[    0.208435] sysfs: cannot create duplicate filename '/devices/platform/omapdss'
[    0.208465] Modules linked in:
[    0.208496] [<c001a030>] (unwind_backtrace+0x0/0xf0) from [<c004d818>] (warn_slowpath_common+0x4c/0x64)
[    0.208557] [<c004d818>] (warn_slowpath_common+0x4c/0x64) from [<c004d8c4>] (warn_slowpath_fmt+0x30/0x40)
[    0.208587] [<c004d8c4>] (warn_slowpath_fmt+0x30/0x40) from [<c014834c>] (sysfs_add_one+0x90/0xb8)
[    0.208618] [<c014834c>] (sysfs_add_one+0x90/0xb8) from [<c01483d4>] (create_dir+0x60/0xc4)
[    0.208679] [<c01483d4>] (create_dir+0x60/0xc4) from [<c01484e8>] (sysfs_create_dir+0x84/0xcc)
[    0.208709] [<c01484e8>] (sysfs_create_dir+0x84/0xcc) from [<c029f6ec>] (kobject_add_internal+0x9c/0x1f4)
[    0.208740] [<c029f6ec>] (kobject_add_internal+0x9c/0x1f4) from [<c029fb18>] (kobject_add+0x50/0x98)
[    0.208801] [<c029fb18>] (kobject_add+0x50/0x98) from [<c02fc8dc>] (device_add+0x84/0x524)
[    0.208831] [<c02fc8dc>] (device_add+0x84/0x524) from [<c0300524>] (platform_device_add+0xf8/0x1ac)
[    0.208862] [<c0300524>] (platform_device_add+0xf8/0x1ac) from [<c06488a8>] (igep_init+0x1b0/0x28c)
[    0.208923] [<c06488a8>] (igep_init+0x1b0/0x28c) from [<c063c26c>] (customize_machine+0x1c/0x28)
[    0.208953] [<c063c26c>] (customize_machine+0x1c/0x28) from [<c00086f8>] (do_one_initcall+0x34/0x180)
[    0.208984] [<c00086f8>] (do_one_initcall+0x34/0x180) from [<c0639850>] (kernel_init+0x88/0x134)
[    0.209045] [<c0639850>] (kernel_init+0x88/0x134) from [<c00141f8>] (kernel_thread_exit+0x0/0x8)
[    0.209289] ---[ end trace 1b75b31a2719ed1c ]---

Signed-off-by: Enric Balletbo i Serra <eballetbo@iseebcn.com>
---
 arch/arm/mach-omap2/board-igep0020.c |   15 ++-------------
 1 files changed, 2 insertions(+), 13 deletions(-)

diff --git a/arch/arm/mach-omap2/board-igep0020.c b/arch/arm/mach-omap2/board-igep0020.c
index 01de013..139fe35 100644
--- a/arch/arm/mach-omap2/board-igep0020.c
+++ b/arch/arm/mach-omap2/board-igep0020.c
@@ -223,14 +223,6 @@ static struct omap_dss_board_info dss_board_data = {
 	.default_device	= &igep00x0_dvi_device,
 };
 
-static struct platform_device igep0020_dss_device = {
-	.name	= "omapdss",
-	.id	= -1,
-	.dev	= {
-		.platform_data = &dss_board_data,
-	},
-};
-
 static inline void igep0020_display_init(void)
 {
 	if ((gpio_request(GPIO_DVI_PUP, "DVI PUP") == 0) &&
@@ -238,11 +230,6 @@ static inline void igep0020_display_init(void)
 		gpio_export(GPIO_DVI_PUP, 0);
 	else
 		pr_err("IGEP: Could not obtain gpio DVI PUP\n");
-
-	igep00x0_dvi_device.platform_enable = igep0020_enable_dvi;
-	igep00x0_dvi_device.platform_disable = igep0020_disable_dvi;
-
-	platform_device_register(&igep0020_dss_device);
 }
 
 static int igep0020_keymap[] = {
@@ -423,6 +410,8 @@ static void __init igep_init(void)
 	igep00x0_smsc911x_init(SMSC911X_CS, SMSC911X_IRQ, SMSC911X_NRESET);
 
 	/* Display initialitzation */
+	igep00x0_dvi_device.platform_enable = igep0020_enable_dvi;
+	igep00x0_dvi_device.platform_disable = igep0020_disable_dvi;
 	omap_display_init(&dss_board_data);
 	igep0020_display_init();
 
-- 
1.7.4.1

